src/main/java/com/ctc/wstx/dtd/DTDValidator.java

```diff
@@ -254,7 +254,7 @@ public String validateAttribute(String localName, String uri,
             /* [WSTX-190] NPE if we continued (after reported didn't
              *   throw an exception); nothing more to do, let's leave
              */
-            return new String(valueChars, valueStart, valueEnd);
+            return new String(valueChars, valueStart, valueEnd-valueStart);
```

Comment:
Looks good -- but I think we need the same a few lines later, at row 281 too? I can add that after merging.